### ISSUE TITLE
I/O error: operation failed to complete synchronously [Windows7,MinGW]

### ISSUE THREAD

**Issue by CosmicSpaceGoose** [rust-lang/rust#99090]
Program crashing with segfault when reading stdin

### Code

```rust
use std::io;

fn main() {
    let mut buf = String::new();

    io::stdin()
        .read_line(&mut buf)
        .expect("Failed to read line");
}
```

### Output

```
I/O error: operation failed to complete synchronously
Segmentation fault
```

### Version where it works

```
rustc 1.61.0 (fe5b13d68 2022-05-18)
binary: rustc
commit-hash: fe5b13d681f25ee6474be29d748c65adcd91f69e
commit-date: 2022-05-18
host: x86_64-pc-windows-gnu
release: 1.61.0
LLVM version: 14.0.0
```

### Version with regression

```
rustc 1.62.0 (a8314ef7d 2022-06-27)
binary: rustc
commit-hash: a8314ef7d0ec7b75c336af2c9857bfaf43002bfc
commit-date: 2022-06-27
host: x86_64-pc-windows-gnu
release: 1.62.0
LLVM version: 14.0.5
```

Might be important that it's a Windows7 without latest updates and MinGW.
MinGW gonna drop Win7 support 'til end of the year. What is the plan for rust in such case?

**Comment by ChrisDenton** [comment]
This should be fixed by #98950.

The future of Rust's stdlib support for Windows 7 is currently uncertain. It will likely be supported at least until January next year but beyond that is unknown at this time.

**Comment by CosmicSpaceGoose** [comment]
Oh, so that's the fix. Great. I'm fine to close this issue.
